src/d3a/models/strategy/utils.py

```diff
@@ -18,19 +18,21 @@
 from d3a.constants import DEFAULT_PRECISION, RELATIVE_STD_FROM_FORECAST_ENERGY
 
 
-def alter_energy(
-        energy_kWh: float, relative_std: float = RELATIVE_STD_FROM_FORECAST_ENERGY) -> float:
+def compute_altered_energy(
+        energy_kWh: float, relative_std: float = RELATIVE_STD_FROM_FORECAST_ENERGY,
+        random_generator: np.random.Generator = np.random.RandomState()) -> float:
```

Comment:
Hmm, not sure but this will not generate the same results for the same random seed (global Setting) , right?

Reply:
This will not interfere with the global random seed (the one we set with `np.random.seed()`). But at the same time, this will not inherit the global seed (so the energy values will be completely random). The only way I could think of to change this is to refactor the global random seed completely (maybe using a singleton). However, placing such a singleton in `singletons.py` causes circular imports.

Reply:
You could add it to the GlobalConfig (and consequently to the config.py), no circular imports should present then.
However, again, this is probably an overkill. I think it is worth to ask Colin in the ticket whether the user wants to have reproducible results ( I fear that he/she does).

Reply:
Another way to go would be to just store a constant for the seed, and then replace all usages of numpy's `random` to use `np.random.default_rng(seed=constants.SEED)`

Reply:
ok! This would be the discussed additional PR right? IMO it will be worth it (if reproducible results are wanted)

Reply:
> However, again, this is probably an overkill. I think it is worth to ask Colin in the ticket whether the user wants to have reproducible results ( I fear that he/she does).

Sure, I'll ask Colin and if that's the case I'll create another subtask for it then!

Reply:
Update: Colin suggested to add a constant seed for the moment. If necessary, we can add new functionalities later on to allow users to change this seed.

Reply:
Why cant we use the seed we already have in the simulation config?

Reply:
It was discussed offline to use a constant and separate seed for the `compute_altered_energy`.